assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -1,28 +1,25 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
 import I18n from "I18n";
 import { bind } from "discourse-common/utils/decorators";
+import { getOwner } from "discourse-common/lib/get-owner";
 
 export default {
   name: "chat-setup",
   initialize(container) {
-    this.currentUser = container.lookup("current-user:main");
     this.chatService = container.lookup("service:chat");
 
     withPluginApi("0.12.1", (api) => {
       // we want to decorate the chat quote dates regardless
       // of whether the current user has chat enabled
       api.decorateCookedElement(
         (elem) => {
-          if (!this.currentUser) {
-            this.currentUser = container.lookup("current-user:main");
-          }
-
-          const currentUserTimezone = this.currentUser?.resolvedTimezone(
-            this.currentUser
-          );
+          const currentUser = getOwner(this).lookup("current-user:main");
+          const currentUserTimezone =
+            currentUser?.resolvedTimezone(currentUser);
```

Comment:
Not related to this PR but the API here feels weird. We have to pass the `currentUser` as an argument to the function called in the context of `currentUser`

Reply:
> the API here feels weird

Yeah, that's **exactly** what I said when reviewing another PR 😅 https://github.com/discourse/discourse/pull/15749#pullrequestreview-866816280 I suppose it could be turned into a low-pri todo 🙂